src/System.Composition.Hosting/src/System/Composition/Hosting/Util/Formatters.cs

```diff
@@ -31,8 +37,15 @@ public static string Format(Type type)
 
         private static string FormatClosedGeneric(Type closedGenericType)
         {
-            Assumes.NotNull(closedGenericType);
-            Assumes.IsTrue(closedGenericType.IsConstructedGenericType);
+            if (closedGenericType == null)
+            {
+                throw new ArgumentNullException(nameof(closedGenericType));
+            }
+
+            if (!closedGenericType.IsConstructedGenericType)
+            {
+                throw new ArgumentException();
```

Comment:
this should have the generic error message `Diagnostic_InternalExceptionMessage` that you probably need to add to the resource dictionary `SR` in this project.